packages/api/src/db/UserCollection.ts

```diff
@@ -34,6 +34,10 @@ export class UserCollection extends CollectionBase<DbUser> {
 		this.updateItem({ id: user.id }, { $set: { forgot_password_token: token } })
 	}
 
+	public setPreferences(userId: DbUser['id'], preferences: string) {
+		this.updateItem({ id: userId }, { $set: { preferences } })
```

Comment:
can you make it match the approach used in setPasswordResetTokenForUser for the user id?

Specifically have it send in a user as param and then user `{ id: user.id }` in the function.

Reply:
The change has been made.
But I disagree about it, I don't think we have to retrieve the whole User from the DB, to then send it whole to a method when we only need its `id`. That's why I added the collection method `exist` to just check if the `id` provided was legit.